app/controllers/spree/gift_cards_controller.rb

```diff
@@ -45,5 +46,9 @@ def gift_card_params
       params.require(:gift_card).permit(:email, :name, :note, :variant_id)
     end
 
+    def load_variant
+      @current_variant = Spree::Product.friendly.find(params[:id]).try(:variants_including_master).try(:first) if (params[:id]).present?
```

Comment:
@AnuragBhardwaj1 Move Spree::Product.friendly.find(params[:id]) to a separate before_action load_product